test/functional/p2p_invalid_tx.py

```diff
@@ -158,13 +157,71 @@ def run_test(self):
         with node.assert_debug_log(['mapOrphan overflow, removed 1 tx']):
             node.p2p.send_txs_and_test(orphan_tx_pool, node, success=False)
 
+        self.log.info('Test orphan with rejected parents')
         rejected_parent = CTransaction()
         rejected_parent.vin.append(CTxIn(outpoint=COutPoint(tx_orphan_2_invalid.sha256, 0)))
         rejected_parent.vout.append(CTxOut(nValue=11 * COIN, scriptPubKey=SCRIPT_PUB_KEY_OP_TRUE))
         rejected_parent.rehash()
         with node.assert_debug_log(['not keeping orphan with rejected parents {}'.format(rejected_parent.hash)]):
             node.p2p.send_txs_and_test([rejected_parent], node, success=False)
 
+        self.log.info('Test erase of orphan tx from peer')
+        with node.assert_debug_log(['Erased 100 orphan tx from peer=24']):
+            self.reconnect_p2p(num_connections=1)
+
+        self.log.info('Test erase of orphan tx included by block')
+        tx_withhold_until_blockA = CTransaction()
+        tx_withhold_until_blockA.vin.append(CTxIn(outpoint=COutPoint(tx_withhold.sha256, 1)))
+        tx_withhold_until_blockA.vout = [CTxOut(nValue=12 * COIN, scriptPubKey=SCRIPT_PUB_KEY_OP_TRUE)] * 2
+        tx_withhold_until_blockA.calc_sha256()
+
+        tx_orphan_include_by_blockA = CTransaction()
+        tx_orphan_include_by_blockA.vin.append(CTxIn(outpoint=COutPoint(tx_withhold_until_blockA.sha256, 0)))
+        tx_orphan_include_by_blockA.vout.append(CTxOut(nValue=12 * COIN - 12000, scriptPubKey=SCRIPT_PUB_KEY_OP_TRUE))
+        tx_orphan_include_by_blockA.calc_sha256()
+
+        self.log.info('Send the tx_orphan_include_by_blockA ... ')
+        node.p2p.send_txs_and_test([tx_orphan_include_by_blockA], node, success=False)
```

Comment:
This is checking the the txn is not accepted into the mempool; mind mentioning the expected outcome in the info log and why? ISTM this is the assertion for the subtest beginning at line 172? Perhaps just one log at 172 that describes the assertion/expected outcome/why, then one at the start of the next subtest that does the same.

Reply:
Re-worded. It seems mention "orphan" transaction implies that it is not accepted into the mempool, no?